exonum/src/blockchain/service.rs

```diff
@@ -451,6 +451,11 @@ impl SharedNodeState {
         state.is_enabled = is_enabled;
     }
 
+    pub(crate) fn set_validator(&self, is_validator: bool) {
```

Comment:
Naming seems weird for me. Maybe `set_node_role`?

Reply:
in that case, it should receive an enum

Reply:
@ozkriff Is I got it right? `enum NodeRole { Validator, Auditor, }`
If yes, wouldn't it be overcomplicated?
Because we need to match this enum in `set_node_role` and before it to construct this enum we need to match bool value to this enum

Reply:
`is_validator` variable should be replaced with this enum as well